### PR TITLE
Add idealZoom trait to mappable.

### LINKED ISSUE #6119: Support overriding ideal zoom
In some cases, a catalogue item may need to specify its ideal zoom instead of using default extents, viewing angle, position and up.

### PR DESCRIPTION
**Pull request by mwu2018**
### What this PR does

Fixes #6119 
  
### Test me
- Go to dev site. 
- Add `3D Imagery` -> `Anglesea 3D Photomesh`.
- Click on `IDEAL ZOOM`. You should see a river and bridge.

### Checklist

-   [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
-   [ ] I've updated relevant documentation in `doc/`.
-   [x] I've updated CHANGES.md with what I changed.
-   [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/ReactViews/Workbench/Controls/ViewingControls.jsx [hunk 1]
```diff
@@ -134,7 +135,24 @@ const ViewingControls = observer(
...
+          // No value checking here. Improper values can lead to unexpected results.
+          const lookAt = {
+            targetLongitude: item.idealZoom.targetLongitude,
+            targetLatitude: item.idealZoom.targetLatitude,
+            targetHeight: item.idealZoom.targetHeight,
+            heading: item.idealZoom.heading,
+            pitch: item.idealZoom.pitch,
+            range: item.idealZoom.range
+          };
+          zoomToView = CameraView.fromJson({ lookAt: lookAt });
```
Comment (tephenavies) [inline lib/ReactViews/Workbench/Controls/ViewingControls.jsx]: This `else if` clause should be above the rectangle > 360 if block. `idealZoom`, when available, should be used over any other zoom to option

File: test/ReactViews/Workbench/Controls/IdealZoomSpec.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+
+    testRenderer.root.findAllByType("button")[0].props.onClick();
+    const theCameraView = terria.currentViewer.getCurrentCameraView();
+
+    const directionX = 0.6595071845691584;
+    const directionY = -0.37148703631220265;
+    const directionZ = -0.6534888333809832;
+    expect(theCameraView.direction?.x).toBe(directionX);
+    expect(theCameraView.direction?.y).toBe(directionY);
+    expect(theCameraView.direction?.z).toBe(directionZ);
```
Comment (tephenavies) [inline test/ReactViews/Workbench/Controls/IdealZoomSpec.tsx]: Probably should use `.toBeCloseTo(variable, 6)` for floating point comparison (compare with 6 decimal digits of precision - could use anything from 4-12). We've had Cesium updates that change these sorts of results by .00000000000001 and that shouldn't fail the test.

**Review by tephenavies — CHANGES_REQUESTED** [review]

**Comment by tephenavies** [comment]
Tested with: http://ci.terria.io/issue-6119/#clean&https://gist.githubusercontent.com/steve9164/e49a14b40b44fd9030801e1476a661da/raw/f0d3ca4e2ca8ac2bf0677b7cac5b1270839db5df/ideal-zoom.json

Works in 2D Leaflet mode (as well as it can, considering it's a 3D zoom setting)

File: lib/ReactViews/Workbench/Controls/ViewingControls.jsx (hunk 1 above)
Comment (mwu2018) [reply lib/ReactViews/Workbench/Controls/ViewingControls.jsx]: done

File: test/ReactViews/Workbench/Controls/IdealZoomSpec.tsx (hunk 2 above)
Comment (mwu2018) [reply test/ReactViews/Workbench/Controls/IdealZoomSpec.tsx]: done

**Comment by mwu2018** [comment]
> Tested with: http://ci.terria.io/issue-6119/#clean&https://gist.githubusercontent.com/steve9164/e49a14b40b44fd9030801e1476a661da/raw/f0d3ca4e2ca8ac2bf0677b7cac5b1270839db5df/ideal-zoom.json
> 
> Works in 2D Leaflet mode (as well as it can, considering it's a 3D zoom setting)

For 2D viewer, it zooms to rectangle area approximated by the customised camera view parameters. Therefore I only added rectangle value checking, which I think is sufficient.

**Review by tephenavies — APPROVED** [review]

**Pull request merged**